core/blockchain_test.go

```diff
@@ -5077,3 +5079,136 @@ func TestBlockPayloadSizeLimit(t *testing.T) {
 }
 
 // [Scroll: END]
+
+// [Kroma: START]
+func TestValidateDepositTxType(t *testing.T) {
+	zero := uint64(0)
+	one := uint64(1)
+	future := uint64(9999999999)
+	tests := []struct {
+		name         string
+		kromaMPTTime *uint64
+		depositTx    *types.Transaction
+		valid        bool
+	}{
+		{
+			name:         "KromaMPTNotSet_DepositTx",
+			kromaMPTTime: nil,
+			depositTx: types.NewTx(&types.DepositTx{
+				SourceHash: common.HexToHash("0x1234"),
+				Mint:       big.NewInt(34),
+			}),
+			valid: false,
+		},
+		{
+			name:         "KromaMPTNotSet_KromaDepositTx",
+			kromaMPTTime: nil,
+			depositTx: types.NewTx(&types.KromaDepositTx{
+				SourceHash: common.HexToHash("0x1234"),
+				Mint:       big.NewInt(34),
+			}),
+			valid: true,
+		},
+		{
+			name:         "KromaMPTNotActivateYet_DepositTx",
+			kromaMPTTime: &future,
+			depositTx: types.NewTx(&types.DepositTx{
+				SourceHash: common.HexToHash("0x1234"),
+				Mint:       big.NewInt(34),
+			}),
+			valid: false,
+		},
+		{
+			name:         "KromaMPTNotActivateYet_KromaDepositTx",
+			kromaMPTTime: &future,
+			depositTx: types.NewTx(&types.KromaDepositTx{
+				SourceHash: common.HexToHash("0x1234"),
+				Mint:       big.NewInt(34),
+			}),
+			valid: true,
+		},
+		{
+			name:         "KromaMPTAtGenesis_DepositTx",
+			kromaMPTTime: &zero,
+			depositTx: types.NewTx(&types.DepositTx{
+				SourceHash: common.HexToHash("0x1234"),
+				Mint:       big.NewInt(34),
+			}),
+			valid: true,
+		},
+		{
+			name:         "KromaMPTAtGenesis_KromaDepositTx",
+			kromaMPTTime: &zero,
+			depositTx: types.NewTx(&types.KromaDepositTx{
+				SourceHash: common.HexToHash("0x1234"),
+				Mint:       big.NewInt(34),
+			}),
+			valid: false,
+		},
+		{
+			name:         "KromaMPTActivatedAfterGenesis_DepositTx",
+			kromaMPTTime: &one,
+			depositTx: types.NewTx(&types.DepositTx{
+				SourceHash: common.HexToHash("0x1234"),
+				Mint:       big.NewInt(34),
+			}),
+			valid: true,
+		},
+		{
+			name:         "KromaMPTActivatedAfterGenesis_KromaDepositTx",
+			kromaMPTTime: &one,
+			depositTx: types.NewTx(&types.KromaDepositTx{
+				SourceHash: common.HexToHash("0x1234"),
+				Mint:       big.NewInt(34),
+			}),
+			valid: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(st *testing.T) {
+			config := *params.TestChainConfig
+			config.Kroma = &params.KromaConfig{
+				EIP1559Elasticity:        6,
+				EIP1559Denominator:       50,
+				EIP1559DenominatorCanyon: 250,
+			}
+			config.Optimism = &params.OptimismConfig{
+				EIP1559Elasticity:        6,
+				EIP1559Denominator:       50,
+				EIP1559DenominatorCanyon: 250,
+			}
+			config.KromaMPTTime = test.kromaMPTTime
```

Comment:
Shouldn't we add `config.Zktrie = true` unless MPT is activated at genesis?

Reply:
Yeah but isn't it better for consistency?

Reply:
This test includes tests for upgrading from pre KromaMPT to post KromaMPT, so wouldn’t adding that configuration be a bit unclear?